### PR TITLE
Adds a UP check for device identification

### PR DESCRIPTION
**Pull request by kaczmarczyck**
The spec introduced a feature where empty PIN auth parameters are used to trigger user presence checks. This is already checked in the test tool, and now also reflected here.

- [x] Tests pass

#106

### CODE REVIEW

File: src/ctap/mod.rs [hunk 1]
```diff
@@ -342,6 +342,7 @@ where
         if let Some(auth_param) = &pin_uv_auth_param {
             // This case was added in FIDO 2.1.
             if auth_param.is_empty() {
+                let _ = (self.check_user_presence)(cid);
```
Comment (jmichelp) [inline src/ctap/mod.rs:345]: We do a UP check but we don't care about the result. Seems odd.

Comment (kaczmarczyck) [reply src/ctap/mod.rs:345]: The reason is that we return a specific error code right after.

> If authenticator supports either pinUvAuthToken or clientPin features and the platform sends a zero length pinUvAuthParam, wait for user touch and then return either CTAP2_ERR_PIN_NOT_SET if PIN is not set or CTAP2_ERR_PIN_INVALID if PIN has been set.

The reason for this behaviour:

> This is done for the case where multiple authenticators are attached to the platform and the platform wants to enforce pinUvAuthToken semantics, but the user has to select which authenticator to get the pinUvAuthToken from.

Comment (jmichelp) [reply src/ctap/mod.rs:345]: But if the user doesn't validate UP (i.e. timeout) the authenticator hasn't really been selected but yet will return one of the 2 error codes.
I would expect that the authenticator should return that UP timed out in such case. Maybe something to clarify on the spec.

Comment (kaczmarczyck) [reply src/ctap/mod.rs:345]: Discussion opened for clarification on the [spec](https://github.com/fido-alliance/fido-2-specs/issues/1092) (link for those with permissions). I'll keep this PR open until resolved.

Comment (kaczmarczyck) [reply src/ctap/mod.rs]: https://github.com/fido-alliance/fido-2-specs/pull/1096
Actually, the user touch shouldn't be ignored. Spec PR on the way. Changed in the next commit.

**Comment by jmichelp** [comment]
Having a second look, the 2 snippets shown in the diff look identical. Maybe worth a code refactoring to avoid code duplication?

**Comment by kaczmarczyck** [comment]
Done, moved to a function and made it a bit nicer.

**Review by jmichelp — APPROVED** [review]

**Pull request merged**